lib/model_context_protocol/server.rb

```diff
@@ -254,5 +255,21 @@ def index_resources_by_uri(resources)
         hash[resource.uri] = resource
       end
     end
+
+    def method_parameters(method)
+      method_def = method_definition(method)
+      method_def.parameters.flatten
+    end
+
+    def method_definition(method)
+      if defined?(T::Utils) && T::Utils.respond_to?(:signature_for_method)
+        sorbet_method_definition = T::Utils.signature_for_method(method)
```

Comment:
`T::Utils.signature_for_method` returns `T::Private::Methods::Signature` instead of method objects, which also has `parameters` method so it doesn't error. 

```
irb(main):013> T::Utils.signature_for_method(method(:foo)).class
=> T::Private::Methods::Signature
irb(main):014> T::Utils.signature_for_method(method(:foo)).parameters
=> [[:req, :x]]
irb(main):015> T::Utils.signature_for_method(method(:foo)).method.parameters
=> [[:req, :x]]
```

But for clarify, I think it's better to call `method` on it to make sure `sorbet_method_definition` and `method` will both be `Method` objects.

```suggestion
        sorbet_method_definition = T::Utils.signature_for_method(method).method
```

Reply:
Updated, and added additional test coverage to server_test.rb